CHANGELOG.rst

```diff
@@ -1,3 +1,27 @@
+- Fish now asks the terminal to speak the keyboard input protocol CSI u in combination with XTerm's ``modifyOtherKeys`` and Kitty's progressive enhancements.
+  This allows to bind almost any key, including arbitrary combinations of modifiers ctrl, alt and shift.
+- There is a new syntax for specifying keys for builtin ``bind``.
+  The new syntax introduces modifier names and names for some keys that don't have an obvious and printable Unicode code point. The old syntax remains mostly supported but the new one is preferred.
+  - Existing bindings that use the new names have a different meaning now.
+    For example
+    - ``bind up 'do something'`` maps the up arrow instead of the two keys ``u`` and ``p``.
+    - ``bind ctrl-x,alt-c 'do something'`` maps a sequence of two key chords.
+  - To minimize breakage, ``KEYS`` argument to ``bind`` is parsed in the old syntax in two cases:
+    - If ``KEYS`` starts with a raw escape character (`\e`) or a raw ASCII control character (`\c`).
+    - If it contains none of ``,`` or ``-`` and it is not a named key.
+  Since ``,`` and ``-`` act as separators, they need to be given as ``comma`` and ``minus`` in ambiguous cases.
+- Specifying key names as terminfo name (``bind -k``) is deprecated and may be removed in a future version.
+- Legacy bindings that use raw escape sequences instead of named modifiers/keys continue to be included in the default configuration.
+  If your terminal supports CSI u, you can remove legacy bindings (including terminfo bindings) by setting the `fish_legacy_bindings` variable to ``false`` (on shell startup or before switching key bindings).
+- The ``fish_escape_delay_ms`` configuration knob has been obsolete (for a while I think) and has been removed.
+- ``fish_key_reader --verbose`` is now ignored, so it no longer shows codepoint values or timing information (though the latter might be useful for users of ``fish_sequence_key_delay_ms``).
+- When a terminal pastes text into fish using bracketed paste, fish used to switch to a special ``paste`` bind mode.
+  This bind mode has been removed. The behavior on paste is not intended to be configurable.
+- Focus reporting is enabled unconditionally, not just for tmux.
+  To use it, define functions that handle events for``fish_focus_in` and ``fish_focus_out``.
+- When fish is stopped or terminated by a signal that cannot be caught (SIGSTOP or SIGKILL), it may leave the terminal in a state where keypresses with modifiers send CSI u sequences instead of traditional ones (that are recognized by bash). If this happens, you can use the ``reset`` command from ``ncurses`` to restore the terminal state.
```

Comment:
`stty sane` is another thing.

Will return/enter be recognized correctly? E.g. if ICRNL remains set/unset you may need to press ctrl-j to get the shell to run a command.

Reply:
> Will return/enter be recognized correctly? 

yes, normal keys, backspace and enter are always compatible, that's what CSI u says

Reply:
`stty sane` doesn't work here unfortunately. Keys like `ctrl-p` go bad in bash after I SIGKILL its child fish